source/Hazard.cpp

```diff
@@ -60,7 +60,10 @@ void Hazard::Load(const DataNode &node)
 		}
 		else if(key == "environmental effect")
 		{
-			int count = (child.Size() >= 3) ? child.Value(2) : 1;
+			// Fractional counts may be accepted, since the real count gets multiplied by the strength
+			// of the hazard. The resulting real count will then be rounded down to the nearest int
+			// to determine the number of effects that appear.
+			float count = (child.Size() >= 3) ? child.Value(2) : 1.;
```

Comment:
Might not be strictly necessary, but it's done elsewhere.
```suggestion
			float count = (child.Size() >= 3) ? static_cast<float>(child.Value(2)) : 1.f;
```